libbpf-cargo/src/build.rs

```diff
@@ -128,6 +147,9 @@ fn compile_one(debug: bool, source: &Path, out: &Path, clang: &Path, options: &s
 }
 
 fn compile(debug: bool, objs: &[UnprocessedObj], clang: &Path) -> Result<()> {
+    let header_dir = extract_libbpf_headers_to_disk()?;
```

Comment:
Is this going to force users to use the libbpf-sys provided bpf headers?

Reply:
> libbpf-sys provided bpf headers

To clarify, this PR only affects libbpf API headers. Kernel bpf headers remain unaffected.

By default, it will have includes like `#include <bpf/bpf_helpers.h>` resolve to libbpf-sys provided helpers. But because we the libbpf-sys `-I` at the end of the clang args, the user can provide their own search path with `-I` and it should get resolved before libbpf-sys's path.

But I can't think of any case where the user would want to use their own libbpf headers. Kernel bpf headers, sure. But libbpf headers, no.

Reply:
+1